TicTacToe.Foundation/WinningStates/WinningStateFactory.cs

```diff
@@ -0,0 +1,18 @@
+﻿using System.Linq;
+using System.Collections.Generic;
+using TicTacToe.Foundation.Interfaces;
+
+namespace TicTacToe.Foundation.WinningStates
+{
+    public class WinningStateFactory : IWinningStateFactory
+    {
+        public IReadOnlyCollection<IWinningState> CreateWinningStatesCollection(IBoard board)
+        {
+            var columns = Enumerable.Range(0, board.BoardSize).Select(column => new WinningStateColumn(board, column)).Cast<IWinningState>();
```

Comment:
Please, use `.ToList()`.
Please, use more specific `.Select<T>()` to avoid redundant `.Cast<T>()`.